### PR TITLE
[RENTAL_TESTS] Rental tests

### PR DESCRIPTION
**Pull request by OlegTuryk**

### CODE REVIEW

**Review by Zel1oy — APPROVED** [review]
GJ

File: src/test/resources/database/create/add-user-for-repository.sql
Comment (padu12) [inline src/test/resources/database/create/add-user-for-repository.sql:1]: Add empty line

File: src/test/resources/database/delete/delete-data.sql
Comment (padu12) [inline src/test/resources/database/delete/delete-data.sql:1]: Add empty line

**Review by padu12 — CHANGES_REQUESTED** [review]
Nice! Lets improve your tests solution

File: src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        testRentalDto.setRentalDate(testRental1.getRentalDate());
+        testRentalDto.setReturnDate(testRental1.getReturnDate());
+        testRentalDto.setUserId(1L);
+        testRentalDto.setCarId(1L);
+    }
+
+    @Test
+    @DisplayName("Save rental - Valid rental")
+    public void save_ValidRental_ReturnSavedRentalDto() {
+        when(carRepository.findById(1L)).thenReturn(Optional.ofNullable(testCar));
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java]: magic number

File: src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        when(rentalMapper.toResponseDto(testRental1)).thenReturn(testRentalDto);
+        ResponseRentalDto actual = rentalService.save(createRentalRequestDto);
+        assertNotNull(actual);
+        assertEquals(testRentalDto, actual);
+    }
+
+    @Test
+    @DisplayName("Get rental by id - Rental exists")
+    public void getById_RentalExists_ReturnRentalDto() {
+        when(rentalRepository.findById(1L)).thenReturn(Optional.of(testRental1));
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java]: magic number

File: src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        when(rentalMapper.toResponseDto(testRental1)).thenReturn(testRentalDto);
+        when(authenticationService.getUserId()).thenReturn(1L);
+        ResponseRentalDto actual = rentalService.getById(1L);
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java]: magic numbers

File: src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    @Test
+    @DisplayName("Update rental - Rental exists")
+    public void update_RentalExists_ReturnUpdatedRentalDto() {
+        when(rentalRepository.findById(1L)).thenReturn(Optional.of(testRental1));
+        when(rentalRepository.save(any(Rental.class))).thenReturn(testRental1);
+        when(rentalMapper.toResponseDto(testRental1)).thenReturn(testRentalDto);
+
+        ResponseRentalDto actual = rentalService.update(1L);
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java]: magic numbers

File: src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+    @Test
+    @DisplayName("Find all rentals")
+    public void findAll_ReturnAllRentals() {
+        Pageable pageable = PageRequest.of(1, 2);
+        when(rentalRepository.findByUserIdAndIsActive(1L, true, pageable))
+                .thenReturn(List.of(testRental1));
+        when(rentalMapper.toResponseDto(testRental1)).thenReturn(testRentalDto);
+        List<ResponseRentalDto> actual = rentalService
+                .getAllRentalsByUserIdAndState(1L, true, pageable);
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/RentalServiceTest.java]: magic numbers

**Review by padu12 — CHANGES_REQUESTED** [review]

**Review by padu12 — CHANGES_REQUESTED** [review]
And resolve the conflicts, please

**Pull request merged**